core/src/main/java/org/apache/accumulo/core/metadata/schema/RootGcCandidatesJson.java

```diff
@@ -96,20 +98,6 @@ public Stream<String> stream() {
   }
 
   public String toJson() {
-    GSonData gd = new GSonData();
-    gd.candidates = candidates;
-    return GSON.toJson(gd);
-  }
-
-  public static RootGcCandidates fromJson(String json) {
-    GSonData gd = GSON.fromJson(json, GSonData.class);
-
-    Preconditions.checkArgument(gd.version == 1);
-
-    return new RootGcCandidates(gd.candidates);
-  }
-
-  public static RootGcCandidates fromJson(byte[] json) {
-    return fromJson(new String(json, UTF_8));
+    return GSON.toJson(this);
```

Comment:
I'm not so sure about this. The old code was robust against changes to this class. You added a warning above... but that's not as safe as enforcing in code the independence of the JSON serialization from changes to this class.

Reply:
Yeah, I wasn't sure about this this as well. I could make the static method take the object. Or maybe if I drop the empty arg constructor, this won't look so suspect.

Reply:
I brought back the static inner classes in [afac496](https://github.com/apache/accumulo/pull/2718/commits/afac49674106a50c19c142204733652f6e77a13a) but made the names a little more descriptive.